### PR TITLE
Relocate discovery probes

### PR DESCRIPTION
**Pull request by gildub**
https://github.com/ManageIQ/manageiq-network_discovery/issues/8

### CODE REVIEW

**Comment by gildub** [comment]
@blomquisg could you please assign?

**Comment by gildub** [comment]
@bdunne, since you are involved maybe you could help with the review or you might know who to assign it to?

File: lib/manageiq-providers-scvmm.rb [hunk 1]
```diff
@@ -1,2 +1,10 @@
 require "manageiq/providers/scvmm/engine"
 require "manageiq/providers/scvmm/version"
+
+module ManageIQ
+  module Providers
+    module Microsoft
+      autoload :Discovery, 'manageiq/providers/microsoft/discovery'
```
Comment (bdunne) [inline lib/manageiq-providers-scvmm.rb]: I don't think we need this since `app/models` should already be on the autoload path.

Comment (gildub) [reply lib/manageiq-providers-scvmm.rb]: Right, thanks

File: app/models/manageiq/providers/microsoft/discovery.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+        SCVMM_PORTS = MSWIN_PORTS + [
+          8100 # VMM HTTP Console (WCF)
+        ].freeze
+
+        VIRTUAL_SERVER_PORT = 5900 # Microsoft Virtual Machine Remote Control Client
+
+        def self.probe(ost)
+          ost.hypervisor << :msvirtualserver if ManageIQ::Network::Port.open?(ost, VIRTUAL_SERVER_PORT)
+          ost.hypervisor << :scvmm if ManageIQ::Network::Port.scan_open(ost, SCVMM_PORTS).length == 3
```
Comment (bdunne) [inline app/models/manageiq/providers/microsoft/discovery.rb]: Should this be using `all_open?`?

File: app/models/manageiq/providers/microsoft/discovery.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          ost.os << :mswin  if ManageIQ::Network::Port.scan_open(ost, MSWIN_PORTS).length == 2
```
Comment (bdunne) [inline app/models/manageiq/providers/microsoft/discovery.rb]: same

File: app/models/manageiq/providers/microsoft/discovery.rb (hunk 2 above)
Comment (gildub) [reply app/models/manageiq/providers/microsoft/discovery.rb]: Yes, thanks

File: app/models/manageiq/providers/microsoft/discovery.rb (hunk 3 above)
Comment (gildub) [reply app/models/manageiq/providers/microsoft/discovery.rb]: Same, done.

**Comment by bdunne** [comment]
@agrare Can you take a look?

**Comment by gildub** [comment]
@miq-bot add_label gaprindashvili/yes

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 774e353703937967e5dd730bf2ceefdb51716498
Author: Adam Grare <agrare@redhat.com>
Date:   Mon Feb 12 10:22:06 2018 -0500

    Merge pull request #61 from gildub/move-discovery-probe
    
    Relocate discovery probes
    (cherry picked from commit 03e4d760511f4390523c0498d3fee61a953e31d5)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1552665
```

**Pull request merged**